oauth2client/contrib/gce.py

```diff
@@ -85,32 +140,109 @@ class AppAssertionCredentials(AssertionCredentials):
     information to generate and refresh its own access tokens.
     """
 
-    @util.positional(2)
-    def __init__(self, scope='', **kwargs):
+    def __init__(self,
+                 scope=None,
+                 service_account_email=None,
+                 **kwargs):
         """Constructor for AppAssertionCredentials
 
         Args:
-            scope: string or iterable of strings, scope(s) of the credentials
-                   being requested. Using this argument will have no effect on
-                   the actual scopes for tokens requested. These scopes are
-                   set at VM instance creation time and won't change.
+            scope:
+                string or iterable of strings, scope(s) of the credentials
+                being requested. Using this argument will have no effect on
+                the actual scopes for tokens requested. These scopes are
+                set at VM instance creation time and won't change.
+            service_account_email:
+                the email for these credentials. This can be used with custom
+                service accounts, or left blank to use the default service
+                account for the instance. The default service account is
+                usually the shared Compute Engine service account.
         """
-        if scope:
-            warnings.warn(_SCOPES_WARNING)
-        # This is just provided for backwards compatibility, but is not
-        # used by this class.
-        self.scope = util.scopes_to_string(scope)
-        self.kwargs = kwargs
-
         # Assertion type is no longer used, but still in the
         # parent class signature.
-        super(AppAssertionCredentials, self).__init__(None)
-        self._service_account_email = None
+        super(AppAssertionCredentials, self).__init__(
+            None, scopes=scope, **kwargs)
 
-    @classmethod
-    def from_json(cls, json_data):
-        data = json.loads(_from_bytes(json_data))
-        return AppAssertionCredentials(data['scope'])
+        self._service_account_info = {}
```

Comment:
Always replace
```
<name> = <value>
if <condition>:
  <name> = <other value>
```
with
```
if <condition>:
  <name> = <other value>
else:
  <name> = <value>
```
. Now obviously this case escapes the strict letter of that rule because instead of `<name> = <other value>` you have `<mutation of value assigned to <name>>`, but the principle still applies. This should be
```
if service_account_email is None:
  self._service_account_info = {}
else:
  self._service_account_info = {'email': service_account_email}
```
.